### PR TITLE
Refactored IPv4CIDR to do proper CIDR validation and matching

### PR DESCRIPTION
**Pull request by digitalfiz**

### CODE REVIEW

**Comment by digitalfiz** [comment]
This relates to #7

File: src/Whitelist/Definition/IPv4CIDR.php [hunk 1]
```diff
@@ -5,11 +5,28 @@
...
  */
-class IPv4CIDR extends IPCIDR
+class IPv4CIDR extends Definition
 {
-	
+    private $regexp = "/^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}(\/[0-9]{1,2})?$/";
+
+    public function validate()
+    {
+        return preg_match($regexp, $this->_definition);
```
Comment (Jalle19) [inline src/Whitelist/Definition/IPv4CIDR.php]: Missing $this

**Comment by Jalle19** [comment]
Can you add a unit test for the 0.0.0.0/0 case?

**Comment by digitalfiz** [comment]
I will add unit tests and use camelCase for the variables tonight when I get home and update the PR with the changes.

**Comment by digitalfiz** [comment]
Updated with proper variable naming and unit tests.

```bash
✔ 16:41 ~/Projects/php-whitelist-check [ master | …1 ] $ phpunit
PHPUnit 5.0.8 by Sebastian Bergmann and contributors.

.......................................                           39 / 39 (100%)

Time: 694 ms, Memory: 17.50Mb

OK (39 tests, 44 assertions)

Generating code coverage report in Clover XML format ... done
```

I also updated the composer.lock file and the .travis.yml for the new phpunit.xml.

Code coverage should now generate properly now too.

**Comment by Jalle19** [comment]
Thanks. Can you squash everything except the Composer and phpunit.xml changes to the first commit? After that it should be good to go.

**Comment by digitalfiz** [comment]
All cleaned up good sir!

**Pull request merged**